esmvaltool/diag_scripts/ocean/diagnostic_maps.py

```diff
@@ -105,11 +150,247 @@ def make_map_plots(
         plt.close()
 
 
+def make_map_contour(
+        cfg,
+        metadata,
+        filename,
+):
+    """
+    Make a simple contour map plot for an individual model.
+
+    Parameters
+    ----------
+    cfg: dict
+        the opened global config dictionairy, passed by ESMValTool.
+    metadata: dict
+        the metadata dictionairy
+    filename: str
+        the preprocessed model file.
+
+    """
+    # Load cube and set up units
+    cube = iris.load_cube(filename)
+    cube = diagtools.bgc_units(cube, metadata['short_name'])
+
+    # Is this data is a multi-model dataset?
+    multi_model = metadata['dataset'].find('MultiModel') > -1
+
+    # Make a dict of cubes for each layer.
+    cubes = diagtools.make_cube_layer_dict(cube)
+
+    # Load image format extention and threshold.thresholds.
+    image_extention = diagtools.get_image_format(cfg)
+
+    # Load threshold/thresholds.
+    plot_details = {}
+    colours = []
+    thresholds = diagtools.load_thresholds(cfg, metadata)
+
+    for itr, thres in enumerate(thresholds):
+        if len(thresholds) > 1:
+            colour = plt.cm.jet(float(itr) / float(len(thresholds) - 1.))
+        else:
+            colour = plt.cm.jet(0)
+        label = str(thres) + ' ' + str(cube.units)
+        colours.append(colour)
+        plot_details[thres] = {'c': colour,
+                               'lw': 1,
+                               'ls': '-',
+                               'label': label}
+
+    linewidths = [1 for thres in thresholds]
+    linestyles = ['-' for thres in thresholds]
+    # Making plots for each layer
+    for layer_index, (layer, cube_layer) in enumerate(cubes.items()):
+        layer = str(layer)
+        # cube_layer = regrid_irregulars(cube_layer)
+        qplt.contour(cube_layer,
```

Comment:
is cube_layer __always__ a 2D cube? otherwise it will fail and you'll need to index it to get a 2D cube

Reply:
The cube layers are created in line 179:
```
    cubes = diagtools.make_cube_layer_dict(cube)
```

The `diagtools.make_cube_layer_dict` ensures that 3D cubes are converted to a list of 2D cubes, so yes. It should always be a 2D cube. (Assuming that the recipe that calls it is sensible!)